### ISSUE TITLE
Installing Arc Menu Fails with error

### ISSUE THREAD

**Issue by Jeremy-TH** [LinxGem33/Arc-Menu#125]
### Problem:
Installing Arc Menu via extensions.gnome.org through the tweak tool installs but shows red 'Error' on the gnome extensions website and an exclamation in the tweak tool "Error Loading Extension".

The looking glass log shows:
**TypeError: app is null**

Not sure if related but interestingly, if I remove the Arc Menu extension, it causes the User Themes extension to crash with: **gnome-shell[14151]: Extension "user-theme@gnome-shell-extensions.gcampax.github.com" had error: Gio.IOErrorEnum: Error opening file /arc-menu@linxgem33.com/stylesheet.css**

Resetting gnome-shell with Alt-F2 **r** allows me to reinstall the User Themes extension without error.

Please note: I removed the full path from the logs to make it more compact, hope this isn't frowned upon. the extension is installing to (and the part I removed): ~/.local/share/gnome-shell/extensions/

This extension seems to at least install for most people so not sure what the deal is. I did disable, but not remove all the other extensions I have to see if one is conflicting but had the same outcome.

Any thoughts would be appreciated.

Thanks for all the hard work on this extension!

### Distribution:
LSB Version:	1.4
Distributor ID:	Arch
Description:	Arch Linux
kernel: 4.11.12-1-ck-ivybridge

Gnome shell version: GNOME Shell 3.24.3

### relevant journalctl output of the extension install:
org.gnome.Shell.desktop[14151]: Archive:  /tmp/IVNA4Y.shell-extension.zip
org.gnome.Shell.desktop[14151]:    creating: /arc-menu@linxgem33.com/locale/tr/
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/fr/LC_MESSAGES/a
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/widgets.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/el/LC_MESSAGES/a
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/menu.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/stylesheet.css
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/pl/LC_MESSAGES/a
org.gnome.Shell.desktop[14151]:    creating: /arc-menu@linxgem33.com/locale/it/
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/uk_UA/LC_MESSAGE
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/schemas/gschemas.compil
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/schemas/org.gnome.shell
org.gnome.Shell.desktop[14151]:    creating: /arc-menu@linxgem33.com/locale/de/LC_MESSAGES/
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/README.md
org.gnome.Shell.desktop[14151]:    creating: /arc-menu@linxgem33.com/locale/tr/LC_MESSAGES/
org.gnome.Shell.desktop[14151]:    creating: /arc-menu@linxgem33.com/media/
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/pt_BR/LC_MESSAGE
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/ru/LC_MESSAGES/a
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/helper.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/media/logo.png
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/prefsWidgets.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/controller.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/tr/LC_MESSAGES/a
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/extension.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/tooltips.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/convenience.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/de/LC_MESSAGES/a
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/secondaryMenu.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/COPYING
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/CHANGELOG.md
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/AUTHORS
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/prefs.js
org.gnome.Shell.desktop[14151]:    creating: /arc-menu@linxgem33.com/locale/it/LC_MESSAGES/
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/locale/it/LC_MESSAGES/a
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/logger.js
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/media/arc-menu-symbolic
org.gnome.Shell.desktop[14151]:   inflating: /arc-menu@linxgem33.com/constants.js
org.gnome.Shell.desktop[14151]:  extracting: /arc-menu@linxgem33.com/media/.keep
org.gnome.Shell.desktop[14151]:  extracting: /arc-menu@linxgem33.com/metadata.json
gnome-shell[14151]: Extension "arc-menu@linxgem33.com" had error: TypeError: app is null
gnome-software-service.desktop[14417]: 16:46:39:0848 GsPluginShellExtensions no app for changed arc-menu@linxgem33.com
gnome-software-service.desktop[14417]: 16:46:39:0848 GsPluginShellExtensions no app for changed arc-menu@linxgem33.com

**Comment by LinxGem33** [comment]
@Jeremy-TH 

could you remove the extension fully and then try install **gir1.2-clutter-1.0**  & **gir1.2-gmenu-3.0** the terminal command is below as i don’t think its installed on your system hence why i think you maybe getting this error below: 

 if it is just carry on and reboot and try install the extension again and if you could report back and let me know if this solved your issue.
##
*error message* 

![arc-menu error3](https://user-images.githubusercontent.com/13478564/28134003-479df2c6-6707-11e7-8a34-c0a3e704729e.png)
##
*terminal commands for required packages*

```
 sudo apt-get install gir1.2-clutter-1.0 
```
```
sudo apt-get install gir1.2-gmenu-3.0
```

kind regards

**Comment by Jeremy-TH** [comment]
@LinxGem33 

Thank you for the quick response.

Unfortunately, these packages do not exist on there own on Arch however the files they are responsible for installing do exist they just don't get put in the same place as Debian/Ubuntu (see below). They are installed by main **clutter** and main **gnome-menus**  packages respectively.

Output of the pacman version of asking who ownes a file:

```bash
for i in $( sudo find /usr/lib -type f \( -name Cally-1.0.typelib -o -name Clutter-1.0.typelib -o -name ClutterGdk-1.0.typelib -o -name ClutterX11-1.0.typelib -o -name GMenu-3.0.typelib \) | cut -c27-); do pacman -Fs $i; done
extra/clutter 1.26.2-1
    usr/lib/girepository-1.0/Clutter-1.0.typelib
extra/clutter 1.26.2-1
    usr/lib/girepository-1.0/ClutterX11-1.0.typelib
extra/gnome-menus 3.13.3+27+g69656ff-1
    usr/lib/girepository-1.0/GMenu-3.0.typelib
extra/clutter 1.26.2-1
    usr/lib/girepository-1.0/ClutterGdk-1.0.typelib
extra/clutter 1.26.2-1
    usr/lib/girepository-1.0/Cally-1.0.typelib
```
### As for the difference in file location mention above:

**gir1.2-clutter-1.0** from Debian/Ubuntu installs:
_/usr/lib/x86_64-linux-gnu/girepository-1.0/Cally-1.0.typelib
/usr/lib/x86_64-linux-gnu/girepository-1.0/Clutter-1.0.typelib
/usr/lib/x86_64-linux-gnu/girepository-1.0/ClutterGdk-1.0.typelib
/usr/lib/x86_64-linux-gnu/girepository-1.0/ClutterX11-1.0.typelib_

**gir1.2-gmenu-3.0** from Debian/Ubuntu installs: 
_/usr/lib/x86_64-linux-gnu/girepository-1.0/GMenu-3.0.typelib_

On Arch however, and as you can see from the above pacman output, these files exist but all reside in:
_/usr/lib/girepository-1.0/_

and just to double check, a quick find yields:

```bash
sudo find / -type f \( -name Cally-1.0.typelib -o -name Clutter-1.0.typelib -o -name ClutterGdk-1.0.typelib -o -name ClutterX11-1.0.typelib -o -name GMenu-3.0.typelib \)
/usr/lib/girepository-1.0/Clutter-1.0.typelib
/usr/lib/girepository-1.0/ClutterX11-1.0.typelib
/usr/lib/girepository-1.0/GMenu-3.0.typelib
/usr/lib/girepository-1.0/ClutterGdk-1.0.typelib
/usr/lib/girepository-1.0/Cally-1.0.typelib
```

I even tried, just for funsies, to create the _x86_64-linux-gnu/girepository-1.0/_ directory and put the 5 aforementioned files there in case it is a path reference problem but no dice, still get the **TypeError: app is null** error. 

I also tried the manual install method with a git clone/make install with the same results.

I'm fully aware that this seems to be a weird one-off perhaps entirely due to my system configuration or simply an Arch thing but any further thoughts on the matter are appreciated.

Please let me know if any additional debugging or terminal outputs would be helpful

Thanks again for the work on this extension.

**Comment by lexruee** [comment]
@Jeremy-TH 
Thanks for the extensive bug report. These are bug reports we like :smile:.

Regarding your issue, I'm must say the following errors are very strange: 
* gnome-shell[14151]: Extension "arc-menu@linxgem33.com" had error: TypeError: app is null
* gnome-software-service.desktop[14417]: 16:46:39:0848 GsPluginShellExtensions no app for changed arc-menu@linxgem33.com

However, on my Arch Notebook and Arch vm I was not able to reproduce this error. Can you reproduce this error on a Arch vm system that "simulates" your current system?

**Comment by Jeremy-TH** [comment]
OK, so I apologize greatly for the noise here. There is something definitely wrong somewhere on my system, I just don't know where.

I threw together an Arch VM and installed a fresh, untouched, unmodified install of gnome and Arc Menu installed just fine.

I threw everything I could think of at this VM:
- Installed all the same extension I have installed on my physical machine, One at a time, removing Arc Menu, Restarting the shell, re-installing Arc Menu
- installed and uninstalled gnome-shell-extensions (a package on Arch that installs some common extensions but puts them in the /usr/share/gnome-shell/extensions directory for global access) which i had installed previously on my physical machine
- removed GDM and installed LightDM (What I use on my physical machine because I had cinnamon installed at one point and just never activated GDM)
- changed, modified, attempted to bork any shell related configurations I could think of to mach my physical machines gnome install
- Plus some stuff I probably did but forgot. 

In short: I put the VM through the wringer and COULD NOT get Arc Menu to fail. It installed perfectly every time. Also removing it through the Tweak tool NEVER once crashed the User Themes extension like it does on my physical machine when I remove Arc Menu through the tweak tool (see my original post)

This is definitely my problem not yours and something I will have to dive deeper into to figure out. 

Thanks again for the time you spent responding and again, i apologize for the noise.

**Comment by LinxGem33** [comment]
@Jeremy-TH 
appreciate you letting us know, i shall close this issue then if you have any issues please don’t hesitate to let us know.

kind regards

**Comment by lexruee** [comment]
@Jeremy-TH 
Since you  are an Arch User, are you willing to test an Arc Menu Arch package? Here is the github repo:
https://github.com/Arc-Menu/gnome-shell-extension-arc-menu-arch

Regarding your installation issue, maybe you're just missing the gnome-shell-extensions package. But full disclaimer, I'm not an Arch-savy Linux user ;-)

**Comment by Jeremy-TH** [comment]
I would love to. Won't be until mid next week as I'm out of town but I will as soon as I'm back. 

I got a shiny new SSD so I have a fresh install of Arch/Gnome and if course Arc Menu works perfectly. I still have my old hard drive so I can still boot it and test which I may mess with a little since I hate not knowing why things that should work don't. 

At any rate, I'll try that package when I get a chance.
